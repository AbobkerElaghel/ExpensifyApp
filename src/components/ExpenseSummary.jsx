import React from "react";
import { connect } from "react-redux";
import numeral from 'numeral';
import getFilteredExpenses from "../selectors/expenses.js";
import getExpensesTotal from "../selectors/getExpensesTotal";
import {Link} from "react-router-dom";

//const ExpenseList =
export const ExpenseSummary = props => (
    <div className="page-header">
        <div className="content-container">
            <h1 className="page-header__title">Viewing <span>{props.count}</span> {props.count === 1 ? "expenses" : "expense"}, totaling <span>{numeral(props.total / 100).format('$0,0.00')}</span></h1>
            <div className="page-header__actions">
                <Link className="button" to="/create">Add Expense</Link>
            </div>
        </div>
    </div>

);

const mapStateToProps = state => {
    const expenses = getFilteredExpenses(state.expenses, state.filters);
    const total = getExpensesTotal(expenses);
    const count = expenses.length;
    return {total, count};
};

export default connect(mapStateToProps)(ExpenseSummary);