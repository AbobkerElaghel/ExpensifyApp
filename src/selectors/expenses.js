import moment from "moment";

// get visible expenses
export default (expenses, {text, sortBy, startDate, endDate}) => (expenses.filter(expense => {
    const createdAtMoment = moment(expense.createdAt);
    const startDateMatch = startDate ? startDate.isSameOrBefore(createdAtMoment) : true;
    const endDateMatch = endDate ? endDate.isSameOrAfter(createdAtMoment) : true;
    const textMatch = expense.description.toLowerCase().includes(text.toLowerCase());
    return startDateMatch && endDateMatch && textMatch;
}).sort((a, b) => {
    if(sortBy === "amount"){
        return b.amount - a.amount;
    } else if (sortBy === "date") {
        return b.createdAt - a.createdAt;
    }}));
