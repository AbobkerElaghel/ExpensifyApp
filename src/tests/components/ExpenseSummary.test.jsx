import React from "react";
import {shallow} from "enzyme";
import {ExpenseSummary} from "../../components/ExpenseSummary.jsx";
import getExpensesTotal from '../../selectors/getExpensesTotal.js';
import expenses from "../fixtures/expenses";

test('should render ExpenseSummary with 0 expenses correctly', () => {
    const wrapper = shallow(<ExpenseSummary total={0} count={0}/>);
    expect(wrapper).toMatchSnapshot();
});

test('should render ExpenseSummary with 0 expenses correctly', () => {
    const wrapper = shallow(<ExpenseSummary total={1} count={69}/>);
    expect(wrapper).toMatchSnapshot();
});

