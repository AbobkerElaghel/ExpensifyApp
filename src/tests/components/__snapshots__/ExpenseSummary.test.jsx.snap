// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseSummary with 0 expenses correctly 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        0
      </span>
       
      expense
      , totaling 
      <span>
        $0.00
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;

exports[`should render ExpenseSummary with 0 expenses correctly 2`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        69
      </span>
       
      expense
      , totaling 
      <span>
        $0.01
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;
